lib/utils/fixEmptyLineBefore.mjs

```diff
@@ -0,0 +1,39 @@
+import { assert } from './validateTypes.mjs';
+
+/** @typedef {import('stylelint').Range} Range */
+/** @typedef {import('postcss').Node} Node */
+/** @typedef {(node: Node, newline: string) => Node} Callback */
+
+/**
+ * fix callback for *-empty-line-before rules
+ * @param {object} o
+ * @param {Node} o.node
+ * @param {string=} o.newline
+ * @param {Callback} o.mutate
+ * @param {Range} [o.range]
+ * @returns {Range}
+ */
+export default ({ node, newline, mutate, range }) => {
+	let count;
+
+	switch (mutate.name) {
+		case 'addEmptyLineBefore':
+			count = !/\r?\n/.test(node.raws.before) ? -2 : -1;
+			break;
+		case 'removeEmptyLinesBefore':
+			count = (node.raws.before.match(/\n/g) || []).length - 1;
```

Comment:
[ask] It doesn't seem that this `removeEmptyLinesBefore` case is executed. Will this be used later?

Reply:
It will if `expectEmptyLineBefore` is false in the `walk*` loop of the rules.